lib/urldata.h

```diff
@@ -201,6 +201,8 @@
 /* Download buffer size, keep it fairly big for speed reasons */
 #undef BUFSIZE
 #define BUFSIZE CURL_MAX_WRITE_SIZE
+#define MAX_BUFSIZE 524288
```

Comment:
I tried 1MB buffer, and anecdotally I think it's diminishing returns.

16KB -> 0.6MB/s
512KB -> 9.4MB/s
1024KB -> 9.9MB/s